tests/utils/test_helpers.py

```diff
@@ -175,6 +179,48 @@ def test_extend_hub_uri():
             assert expected_output == output
 
 
+@pytest.mark.parametrize(
+    "tag_name,raise_on_failure,expected",
+    [
+        (
+            "tag_name",
+            True,
+            pytest.raises(mlrun.errors.MLRunInvalidArgumentError),
+        ),
+        (
+            "tag_with_char!@#",
+            True,
+            pytest.raises(mlrun.errors.MLRunInvalidArgumentError),
+        ),
+        (
+            "tag^name",
+            True,
+            pytest.raises(mlrun.errors.MLRunInvalidArgumentError),
+        ),
+        (
+            "(tagname)",
+            True,
+            pytest.raises(mlrun.errors.MLRunInvalidArgumentError),
+        ),
+        (
+            "tagname%",
+            True,
+            pytest.raises(mlrun.errors.MLRunInvalidArgumentError),
+        ),
+        ("tagname2.0", True, does_not_raise()),
+        ("tag-name", True, does_not_raise()),
+        ("tag-NAME", True, does_not_raise()),
+    ],
+)
+def test_validate_tag_name(tag_name, raise_on_failure, expected):
```

Comment:
same here. Especially since in all the test cases here `raise_on_failure = True` so it has no relevance